Test/Tests/RXPromiseTest.mm

```diff
@@ -3132,6 +3132,221 @@ -(void) testTreeCancel2WithBoundPromise
     }
 }
 
+#pragma mark - progress
+
+- (void) testSimpleProgressReporting {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"Progress should be equal to sum of progresses"];
+    NSArray *progressValues = @[@0.0, @0.25, @0.5, @0.75, @1.0];
+    RXPromise *promise = [[RXPromise alloc] init];
+    __block float progress = 0;
```

Comment:
Here, `progress` will be accessed from different threads. We need to ensure the access is thread-safe, for example ensuring the execution context is set to the main thread, or use a dedicated serial dispatch queue.